priv/repo/migrations/20170419210307_create_user.exs

```diff
@@ -0,0 +1,15 @@
+defmodule RemoteRetro.Repo.Migrations.CreateUser do
+  use Ecto.Migration
+
+  def change do
+    create table(:users) do
+      add :google_user_info, :jsonb
```

Comment:
make sense. the ability to catch any additional fields that google sends in the future is certainly a benefit.

the drawback to this approach alone is that it bypasses the data integrity we could ensure using actual columns. For example, let's say in late 2017 google  changes the "given_name" field to "first_name" (unlikely, but possible). All users who joined prior to that change would see the proper value displayed when `user.google_user_info.given_name` is referenced downstream, but users who join after that change wouldn't see a name, and as developers we'd never know. Using columns with NOT NULL constraints (in conjunction with the unstructured JSONB storage) would alert us when we try to store a value that the google user info no longer provides.

A hybrid approach of makes the most sense to me, as it would provide us with:

  - comprehensive storage of all google data in the unstructured JSONB column
  - data integrity, ensuring that:
    - the fields we reference in the app exist
    - the server would have a catch-point for google changing fields we rely on
  - downstream, we wouldn't have have to reference core attributes at different levels of nesting
    - it would be kinda weird to say the id is 
    - i.e. we can say `user.given_name`, rather than `user.google_user_info.given_name`

The only drawback is some data duplication, but that's a small price to pay.